azurerm/resource_arm_private_dns_cname_record_test.go

```diff
@@ -0,0 +1,348 @@
+  name                = "${azurerm_private_dns_cname_record.test.name}"
+  resource_group_name = "${azurerm_private_dns_cname_record.test.resource_group_name}"
+  zone_name           = "${azurerm_private_dns_cname_record.test.zone_name}"
+  ttl                 = 300
+  record              = "contoso.com"
+}
+`, template)
+}
+
+func testAccAzureRMPrivateDnsCNameRecord_subdomain(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_private_dns_zone" "test" {
+  name                = "acctestzone%d.com"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+}
+
+resource "azurerm_private_dns_cname_record" "test" {
+  name                = "mycnamerecord%d"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  zone_name           = "${azurerm_private_dns_zone.test.name}"
+  ttl                 = 300
+  record              = "test.contoso.com"
+}
+`, rInt, location, rInt, rInt)
+}
+
+func testAccAzureRMPrivateDnsCNameRecord_updateRecords(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_private_dns_zone" "test" {
+  name                = "acctestzone%d.com"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+}
+
+resource "azurerm_private_dns_cname_record" "test" {
+  name                = "mycnamerecord%d"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  zone_name           = "${azurerm_private_dns_zone.test.name}"
+  ttl                 = 300
+  record              = "contoso.com"
+}
+`, rInt, location, rInt, rInt)
+}
+
+func testAccAzureRMPrivateDnsCNameRecord_withTags(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_private_dns_zone" "test" {
+  name                = "acctestzone%d.com"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+}
+
+resource "azurerm_private_dns_cname_record" "test" {
+  name                = "mycnamerecord%d"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  zone_name           = "${azurerm_private_dns_zone.test.name}"
+  ttl                 = 300
+  record              = "contoso.com"
+
+  tags = {
+    environment = "Production"
+    cost_center = "MSFT"
+  }
+}
+`, rInt, location, rInt, rInt)
+}
+
+func testAccAzureRMPrivateDnsCNameRecord_withTagsUpdate(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_private_dns_zone" "test" {
+  name                = "acctestzone%d.com"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+}
+
+resource "azurerm_private_dns_cname_record" "test" {
+  name                = "mycnamerecord%d"
```

Comment:
we have a cleanup tool that works based off the prefix `acctest` - as such could we make this:

```suggestion
  name                = "acctestcname%d"
```